scalardb/test/scalardb/transfer_test.clj

```diff
@@ -0,0 +1,271 @@
+(ns scalardb.transfer-test
+  (:require [clojure.test :refer :all]
+            [jepsen.client :as client]
+            [jepsen.checker :as checker]
+            [scalardb.core :as scalar]
+            [scalardb.transfer :as transfer]
+            [spy.core :as spy])
+  (:import (com.scalar.database.api DistributedTransaction)
+           (com.scalar.database.api Get
+                                    Put
+                                    Result)
+           (com.scalar.database.io IntValue
+                                   Key)
+           (com.scalar.database.exception.transaction CommitException
+                                                      CrudException
+                                                      UnknownTransactionStatusException)
+           (java.util Optional)))
+
+(def ^:dynamic test-records (atom {0 0 1 0 2 0 3 0 4 0}))
+
+(def ^:dynamic get-count (atom 0))
+(def ^:dynamic put-count (atom 0))
+(def ^:dynamic commit-count (atom 0))
+
+(defn- key->id
+  [^Key k]
+  (-> k .get first .get))
+
+(defn- mock-result [id]
+  (reify
+    Result
+    (getValue [this column]
+      (->> (@test-records id)
+           (IntValue. column)
+           Optional/of))))
+
+(defn- mock-get
+  [^Get g]
+  (let [id (-> g .getPartitionKey key->id)]
+    (swap! get-count inc)
+    (Optional/of (mock-result id))))
+
+(defn- mock-put
+  [^Put p]
+  (let [id (-> p .getPartitionKey key->id)
+        [_ v] (-> p .getValues first)]
+    (swap! put-count inc)
+    (swap! test-records #(assoc % id (.get v)))))
+
+(def mock-transaction
+  (reify
+    DistributedTransaction
+    (^Optional get [this ^Get g] (mock-get g))
+    (^void put [this ^Put p] (mock-put p))
+    (^void commit [this] (swap! commit-count inc))))
+
+(def mock-transaction-throws-exception
```

Comment:
This is the same as `mock-transaction-throws-exception`  on line 72?

Reply:
Removed! Thank you!